### ISSUE TITLE
Invalid reference format (missing sha prefix on digest) when building with ko

### ISSUE THREAD

**Issue by danistrebel** [GoogleContainerTools/skaffold#8852]
### Expected behavior

ko build creates an image reference with `:TAG@sha256DIGEST` version indication

### Actual behavior

ko build creates an image reference with `:TAG@DIGEST` where the DIGEST doesn't start with a sha prefix

### Information

- Skaffold version: v2.5.0 
- Operating system: macOS 13.4 
- Installed via: skaffold.dev 
- Contents of skaffold.yaml (ko example + -v flag):

```yaml
apiVersion: skaffold/v4beta5
kind: Config
build:
  artifacts:
  - image: skaffold-ko
    ko:
      flags:
        - -v
      dependencies:
        paths:
        - "**/*.go"
        - go.*
      fromImage: gcr.io/distroless/static-debian11:nonroot
profiles:
- name: gcb
  build:
    googleCloudBuild:
      koImage: us-central1-docker.pkg.dev/k8s-skaffold/testing/skaffold:v1.37.2
```

### Steps to reproduce the behavior

1. clone this repo
2. `cd exammples/ko`
3. set a docker repo in skaffold.env e.g. `SKAFFOLD_DEFAULT_REPO=europe-west1-docker.pkg.dev/my-project/ko-test`
4. `skaffold dev`

Logs

```txt
WARN[0008] error adding artifact to cache; caching may not work as expected: parsing reference "europe-west1-docker.pkg.dev/my-project/ko-test/skaffold-ko:v2.5.0-5-g97de7dba7-dirty@a2c9160597f329f43c02029e47d5da2b7de3d9967e36b62a008064ccb4d4c1ca": invalid reference format  subtask=skaffold-ko task=Build
```

**Comment by ievgenii-shepeliuk** [comment]
The bug was introduced after v2.3.3, i.e. v2.4.1 already has this bug.

**Comment by eshepelyuk** [comment]
> The bug was introduced after v2.3.3, i.e. v2.4.1 already has this bug.

Well Looks like the bug appeared just after 2.x release. 
So 2.x is unusable for `KO builds.

**Comment by danistrebel** [comment]
I tried with v2.3.1 which seems to work fine. After the [switch to ko 0.13 in 2.3.2](https://github.com/GoogleContainerTools/skaffold/compare/v2.3.1...v2.3.2#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R70) I can reproduce the issue again.

Could it be that this commit https://github.com/GoogleContainerTools/skaffold/commit/d2194986f059adf3326c8666c22b4bf530bcc358 have fixed it? If yes, could this be on the changelog for 2.5.2.

**Comment by n3ph** [comment]
Confirming. This bug is fixed in `2.6.0`
